sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c

```diff
@@ -184,8 +210,8 @@ gnrc_pktsnip_t *_dio_prefix_info_build(gnrc_pktsnip_t *pkt, gnrc_rpl_dodag_t *do
     if (_get_pl_entry(dodag->iface, &dodag->dodag_id, prefix_info->prefix_len,
                       &ple)) {
         uint32_t now = (xtimer_now_usec64() / US_PER_MS) & UINT32_MAX;
-        prefix_info->valid_lifetime = byteorder_htonl((ple.valid_until - now) / MS_PER_SEC);
-        prefix_info->pref_lifetime = byteorder_htonl((ple.pref_until - now) / MS_PER_SEC);
+        prefix_info->valid_lifetime = byteorder_htonl(_ms_to_sec(ple.valid_until - now));
+        prefix_info->pref_lifetime = byteorder_htonl(_ms_to_sec(ple.pref_until - now));
```

Comment:
This breaks for infinite lifetime, as `ple.valid_until - now` is checked for equal `UINT32_MAX`, instead of checking `ple.valid_until == UINT32_MAX`.

Reply:
Oh, you're right!! Will fix.

Reply:
Done